src/genie/libs/parser/slxos/show_run.py

```diff
@@ -0,0 +1,752 @@
+                    }
+                }
+            }
+        }
+    }
+
+
+# ===================================
+# Parser for:
+#   * 'show run mac access-list'
+#   * 'show run mac access-list {acl_type}'
+#   * 'show run mac access-list {acl_type} {acl_name}'
+# ===================================
+class ShowRunMacAccessList(ShowRunMacAccessListSchema):
+    ''' Parser for 
+        "show run mac access-list"
+        "show run mac access-list {acl_type}"
+        "show run mac access-list {acl_type} {acl_name}"
+    '''
+
+    cli_command = [
+        'show run mac access-list',
+        'show run mac access-list {acl_type}',
+        'show run mac access-list {acl_type} {acl_name}'
+    ]
+
+    def cli(self, acl_type=None, acl_name=None, output=None):
+
+        if output is None:
+            if acl_type:
+                if acl_name:
+                    cmd = self.cli_command[2].format(acl_type=acl_type, acl_name=acl_name)
+                else:
+                    cmd = self.cli_command[1].format(acl_type=acl_type)
+            else:
+                cmd = self.cli_command[0]
+
+            # get output from device
+            out = self.device.execute(cmd)
+        else:
+            out = output
+
+        res_dict = {}
+
+        # mac access-list extended test3
+        # mac access-list standard test-1
+        p1 = re.compile(
+            r'^mac access-list (?P<acl_type>\w+) (?P<acl_name>\S+)$')
+
+        # seq 10 permit host aaaa.aaaa.aaaa count
+        # seq 20 deny aaaa.aaaa.aaa0 ffff.ffff.fff0 log
+        # seq 30 hard-drop any copy-sflow
+        # seq 103 permit host abcd.abcd.abcd any vlan-tag-format double-tagged outer-vlan 200 inner-vlan 1234 arp
+        # seq 110 permit host abcd.abcd.abcd any vlan-tag-format single-tagged vlan 321 arp
+        # seq 200 permit host abcd.abcd.abcd any vlan-tag-format single-tagged vlan 10 pcp 1 pcp-force 3 drop-precedence-force 2 count log mirror
+        p2 = re.compile(
+            r'^seq (?P<seq_id>\d+) (?P<action>permit|deny|hard-drop) '
+            r'(?P<src>any|(host (([a-fA-F\d]{4}\.){2}[a-fA-F\d]{4})|(([a-fA-F\d]{4}\.){2}[a-fA-F\d]{4}) (([a-fA-F\d]{4}\.){2}[a-fA-F\d]{4})))'
+            r'(?P<dst> (any|(host (([a-fA-F\d]{4}\.){2}[a-fA-F\d]{4})|(([a-fA-F\d]{4}\.){2}[a-fA-F\d]{4}) (([a-fA-F\d]{4}\.){2}[a-fA-F\d]{4}))))'
+            r'?( vlan-tag-format (single-tagged vlan ((?P<vlan_tag_single>\d{1,4})( (?P<vlan_tag_single_mask>0x[a-fA-F0-9]{3}))?)|(double-tagged outer-vlan ((?P<vlan_tag_double_outer>\d{1,4})( (?P<vlan_tag_double_outer_mask>0x[a-fA-F0-9]{3}))?|any) inner-vlan ((?P<vlan_tag_double_inner>\d{1,4})( (?P<vlan_tag_double_inner_mask>0x[a-fA-F0-9]{3}))?|any))))'
+            r'?( vlan (?P<vlan>\d{1,4}))?(?P<ethertype> (arp|ipv4|ipv6|\d{4,5}))'
+            r'?(?P<flags>( log| copy-sflow| arp-guard| count| known-unicast-only| unknown-unicast-only| mirror| drop-precedence-force (?P<drop_precedence_force>\d)| pcp (?P<pcp>\d)| pcp-force (?P<pcp_force>\d))*)?$')
```

Comment:
On one hand, I really like that you tried to match multiple lines of device output with one pattern! However this regex is complicated and hard to read/debug. Would you be willing to split it up into smaller chunks with simpler patterns? This will mean more patterns (and more if blocks in the loop), but each piece will be easier to work on if things ever need to be fixed/reworked

For example one of the patterns could be 
```
# seq 110 permit host abcd.abcd.abcd any vlan-tag-format single-tagged vlan 321 arp
p5 = re.compile("^seq (?P<seq_id>\d+) (?P<action>permit|deny|hard-drop) (?P<src>any|(host \S+))\s+(?P<dst>any|(host \S+))\s+(vlan-tag-format\s+(?P<vlan_tag_format>\S+))\s+vlan\s(?P<vlan>\d+)\s+(?P<ethertype>arp|ipv4|ipv6|\d{4,5})$")
```

What do you think?

Reply:
Note: Since things are working with the regex/logic that you currently have, this isn't a required change